components/src/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import ReactDatePicker from "react-datepicker";
+import { format, isValid } from "date-fns";
+import { debounce } from "throttle-debounce";
+
+import TimePickerInput from "./TimePickerInput";
+import { TimePickerStyles } from "./TimePickerStyles";
+import { Field } from "../Form";
+import { InlineValidation } from "../Validation";
+import { InputFieldPropTypes, InputFieldDefaultProps } from "../Input/InputField.type";
+
+const DEFAULT_TIME_FORMAT = "hh:mm aa";
+const DEFAULT_PLACEHOLDER = "HH:MM";
+
+const getHours = value => {
+  const hours = Number(value.split(":")[0].replace(/\D/g, ""));
+  if (value.includes("PM") && hours < 12) {
+    return hours + 12;
+  }
+  return hours;
+};
+
+const getMinutes = value => {
+  const minutes = value.split(":")[1].replace(/\D/g, "");
+  if (minutes < 60 && minutes.length === 2) {
+    return minutes;
+  }
+  return null;
+};
+
+export const getNewDateWithTime = value => {
+  if (value.length > 4) {
+    const hours = getHours(value);
+    const minutes = getMinutes(value);
+    let date;
+    if (hours < 24 && hours > 0) {
+      date = new Date();
+      date.setHours(hours);
+      date.setMinutes(minutes);
+      return date;
+    }
+    return null;
+  }
+};
+
+class TimePicker extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { selectedTime: props.selected };
+  }
+
+  setSelectedTime = date => {
+    if (isValid(date)) {
+      this.setState(
+        {
+          selectedTime: date
+        },
+        () => this.handleSelectedDateChange(date)
+      );
+    }
+  };
+
+  setSelectedTimeOnInputChange = debounce(1000, value => this.setSelectedTime(getNewDateWithTime(value)));
```

Comment:
how short in duration can this be? the delay felt quite noticeable to me when i tried typing it. i reduced it to 200 and it felt better, but i'm not sure if that could cause an issue or if it could be even faster.

Reply:
I added the delay because otherwise it starts formatting while typing and it gets pretty confusing when you're backspacing, as it starts changing the input value before you're done typing.

Reply:
ah yeah i was just able to replicate that pretty easily :(